service/src/test/java/uk/gov/hmcts/reform/fpl/service/ManagingOrganisationServiceTest.java

```diff
@@ -0,0 +1,131 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.NullSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import uk.gov.hmcts.reform.ccd.model.ChangeOrganisationRequest;
+import uk.gov.hmcts.reform.ccd.model.OrganisationPolicy;
+import uk.gov.hmcts.reform.fpl.exceptions.CaseNotOutsourcedException;
+import uk.gov.hmcts.reform.fpl.exceptions.OrganisationNotFound;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.rd.model.Organisation;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.ccd.model.ChangeOrganisationApprovalStatus.APPROVED;
+import static uk.gov.hmcts.reform.ccd.model.OrganisationPolicy.organisationPolicy;
+import static uk.gov.hmcts.reform.fpl.enums.CaseRole.EPSMANAGING;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.caseRoleDynamicList;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class ManagingOrganisationServiceTest {
+
+    private static final String ORGANISATION_ID = "ORG1";
+
+    @Mock
+    private Time time;
+
+    @Mock
+    private OrganisationService organisationService;
+
+    @InjectMocks
+    private ManagingOrganisationService underTest;
+
+    @Test
+    void shouldReturnManagingOrganisation() {
```

Comment:
could use the @Nested class for each method

Reply:
@Nested does not work well with @MethodSource. @MethodSource requires static method on a class. Cannot add static method to nested non static class. Option is to add
@TestInstance(TestInstance.Lifecycle.PER_CLASS):
but then cannot reuse existing static method in different nested classes